raitools/raitools/raianalyzer/rai_analyzer.py

```diff
@@ -0,0 +1,266 @@
+        meta = {_TARGET_COLUMN: self.target_column,
+                _TASK_TYPE: self.task_type}
+        with open(top_dir / _META_JSON, 'w') as file:
+            json.dump(meta, file)
+        if self._serializer is not None:
+            model_data = self._serializer.save(self.model)
+            # save the serializer
+            with open(top_dir / _SERIALIZER, 'wb') as file:
+                pickle.dump(self._serializer, file)
+            # save the model
+            self._write_to_file(top_dir / _MODEL_PKL, model_data)
+        else:
+            has_setstate = hasattr(self.model, '__setstate__')
+            has_getstate = hasattr(self.model, '__getstate__')
+            if not (has_setstate and has_getstate):
+                raise ValueError(
+                    "Model must be picklable or a custom serializer must"
+                    " be specified")
+            with open(top_dir / _MODEL_PKL, 'wb') as file:
+                pickle.dump(self.model, file)
+
+    @staticmethod
+    def load(path):
+        """Load the RAIAnalyzer from the given path.
+
+        :param path: The directory path to load the RAIAnalyzer from.
+        :type path: str
+        """
+        # create the RAIAnalyzer without any properties using the __new__
+        # function, similar to pickle
+        inst = RAIAnalyzer.__new__(RAIAnalyzer)
+        top_dir = Path(path)
+        # load current state
+        with open(top_dir / (_TRAIN + _DTYPES), 'r') as file:
+            types = json.load(file)
+        with open(top_dir / _TRAIN, 'r') as file:
+            train = pd.read_json(file, dtype=types)
+        inst.__dict__[_TRAIN] = train
+        with open(top_dir / (_TEST + _DTYPES), 'r') as file:
+            types = json.load(file)
+        with open(top_dir / _TEST, 'r') as file:
+            test = pd.read_json(file, dtype=types)
+        inst.__dict__[_TEST] = test
+        with open(top_dir / _META_JSON, 'r') as meta_file:
+            meta = meta_file.read()
+        meta = json.loads(meta)
+        target_column = meta[_TARGET_COLUMN]
+        inst.__dict__[_TARGET_COLUMN] = target_column
+        inst.__dict__[_TASK_TYPE] = meta[_TASK_TYPE]
+        inst.__dict__['_' + _CLASSES] = train[target_column].unique()
+        serializer_path = top_dir / _SERIALIZER
+        if serializer_path.exists():
+            with open(serializer_path) as file:
+                serializer = pickle.load(file)
+            inst.__dict__['_' + _SERIALIZER] = serializer
+            with open(top_dir / _MODEL_PKL, 'rb') as file:
+                inst.__dict__[_MODEL] = serializer.load(file)
+        else:
+            inst.__dict__['_' + _SERIALIZER] = None
+            with open(top_dir / _MODEL_PKL, 'rb') as file:
+                inst.__dict__[_MODEL] = pickle.load(file)
+        # load each of the individual managers
+        managers = []
+        for prop_name, prop_type in dict(vars(RAIAnalyzer)).items():
```

Comment:
So this is at least consistently using reflection, but.... is it really simpler than just having
```python
self._explanation_manager = ExplanationManager.load(....)
```
?

I really feel that this obfuscates what's going on, when listing out each manager load explicitly would be clearer.

Reply:
you mean inst.__dict__['_explanation_manager'] = ExplanationManager.load(top_dir/manager_name, inst)
I feel like it's just as many lines of code (10 here, that would probably also be 10 lines), but this is more maintainable and reduces code duplication.  Also means if we add or remove more managers we don't have to modify this code.

Reply:
You're right, it would be
```python
inst._explanation_manager = ExplanationManager.load(...)
```
There would indeed be one of those for each manager.... five right now, I believe? Personally, I'd prefer the explicit listing, rather than doing reflection on the properties, which also ends up implicitly imposing a naming convention on the managers and properties.

Reply:
done, changed code to call managers explicitly instead of using the code with reflection